tests/test_pr923_fixes.py

```
@@ -0,0 +1,285 @@
+    import tasks.service as svc_mod
+    src = inspect.getsource(svc_mod.TaskExecutionCoordinator.execute)
+    # Find the asyncio.TimeoutError except block
+    assert "asyncio.TimeoutError" in src
+    # The timeout handler must call _requeue_or_block_unavailable, NOT
+    # self.workflow.transition(... TaskStatus.FAILED ...)
+    timeout_block_start = src.index("except asyncio.TimeoutError:")
+    # Find the next except or finally after the timeout block
+    remaining = src[timeout_block_start:]
+    lines = remaining.split("\n")
+    block_lines = []
+    for i, line in enumerate(lines[1:], 1):
```

Ruff (0.15.20)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)